client/client-lib/src/mint/backup.rs

```diff
@@ -272,19 +278,28 @@ impl MintClient {
             end_epoch, "Recovering from snapshot"
         );
 
-        // Since fetching epochs will be slow, we start a dedicated task to do it
-        let (tx, mut rx) = mpsc::channel(10);
-        let self_clone = self.clone();
-        task_group
-            .spawn("fetch epochs", {
-                let epoch_range = epoch_range.clone();
-                |task_handle| async move {
-                    self_clone
-                        .fetch_epochs(epoch_range, tx, &task_handle)
-                        .await
-                }
-            })
-            .await;
+        // Since fetching epochs has high latency, we start a handful dedicated task to do it.
+        // Each will fetch epochs where `epoch - epoch_range.start % num_threads == task_i`,
+        // so task 0 returns epoch first, then 1, ... and so on.
+        let num_threads = 8;
+
+        let mut rx_channels = vec![];
+        for task_i in 0..num_threads {
+            let (tx, rx) = mpsc::channel(1);
+            let self_clone = self.clone();
+            task_group
+                .spawn("fetch epochs", {
+                    let epoch_range = epoch_range.clone();
+                    move |task_handle| async move {
+                        self_clone
+                            .fetch_epochs(epoch_range, num_threads, task_i, tx, &task_handle)
+                            .await
+                    }
+                })
+                .await;
+
+            rx_channels.push(rx);
+        }
```

Comment:
Instead of statically dispatching epochs to specific threads, creating a stream of epoch fetch futures and then calling [`buffered`](https://docs.rs/futures/0.3.26/futures/stream/trait.StreamExt.html#method.buffered) or [`buffer_unordered`](https://docs.rs/futures/0.3.26/futures/stream/trait.StreamExt.html#method.buffer_unordered) with the intended parallelism as argument seems way more elegant.

Reply:
Oh, yeah. I just wasn't familiar with these and short on time, so did it "manually". I'll have to look into it and educate myself a bit.